crates/proto/Cargo.toml

```diff
@@ -98,7 +98,7 @@ tokio = { version = "1.0", features = ["io-util"], optional = true }
 tokio-native-tls = { version = "0.3.0", optional = true }
 tokio-openssl = { version = "0.6.0", optional = true }
 tokio-rustls = { version = "0.22", optional = true, features = ["early-data"] }
-url = "2.1.0"
+url = { version = "2.1.0", features = ["serde"] }
```

Comment:
This should be added to the `serde-config` feature, rather than enabling in all cases I think.

Reply:
Yes, I wanted to do that, however I couldn't figure out how. https://github.com/rust-lang/cargo/issues/1197 made me think it may not be possible?

Reply:
I think that's specific to target feature sets (i.e. linux vs. windows).

replace the `serde-config` feature with this: `serde-config = ["serde", "url/serde"]` in this Cargo.toml, that should work.